EndlessClient/Dialogs/BarberDialog.cs

```diff
@@ -0,0 +1,223 @@
+        private readonly IEIFFileProvider _eifFileProvider;
+        private readonly IEnumerable<ISoundNotifier> _soundNotifiers;
+        private const byte BuySellSfxId = 26;
+
+        private CharacterRenderProperties RenderProperties => _characterControl.RenderProperties;
+        private ListDialogItem _changeHairItem, _changeHairColor, _changeBuyHairStyleOrColor;
+
+        public BarberDialog(INativeGraphicsManager nativeGraphicsManager,
+                            ICharacterRendererFactory rendererFactory,
+                            IEODialogButtonService dialogButtonService,
+                            ICharacterRepository characterLvRepository,
+                            IEODialogIconService dialogIconService,
+                            ILocalizedStringFinder localizedStringFinder,
+                            IBarberActions barberActions,
+                            ICharacterInventoryProvider characterInventoryProvider,
+                            IEOMessageBoxFactory messageBoxFactory,
+                            IEIFFileProvider eifFileProvider,
+                            IEnumerable<ISoundNotifier> soundNotifiers)
+            : base(nativeGraphicsManager, isInGame: true)
+        {
+            _characterRepository = characterLvRepository;
+            _dialogIconService = dialogIconService;
+            _localizedStringFinder = localizedStringFinder;
+            _barberActions = barberActions;
+            _characterInventoryProvider = characterInventoryProvider;
+            _messageBoxFactory = messageBoxFactory;
+            _eifFileProvider = eifFileProvider;
+            _soundNotifiers = soundNotifiers;
+
+            BackgroundTexture = GraphicsManager.TextureFromResource(GFXTypes.PostLoginUI, 56);
+            _characterControl = new CreateCharacterControl(rendererFactory)
+            {
+                DrawPosition = new Vector2(210, 20)
+            };
+
+            InitializeCharacterControl();
+            InitializeDialogItems(dialogButtonService);
+            CenterInGameView();
+        }
+
+        private void InitializeCharacterControl()
+        {
+            var currentProperties = _characterRepository.MainCharacter.RenderProperties;
+            _characterControl.SetParentControl(this);
+            _characterControl.UpdateRenderProperties(currentProperties.HairStyle, currentProperties.HairColor,
+                                                     currentProperties.Race, currentProperties.Gender,
+                                                     currentProperties.ArmorGraphic, currentProperties.WeaponGraphic,
+                                                     currentProperties.BootsGraphic, currentProperties.ShieldGraphic);
+        }
+
+        private void InitializeDialogItems(IEODialogButtonService dialogButtonService)
+        {
+            var cancel = CreateButton(dialogButtonService, new Vector2(215, 150), SmallButton.Cancel);
```

Comment:
I think y coordinate needs to be +1 here
![image](https://github.com/ethanmoffat/EndlessClient/assets/8301078/8741d2e5-8759-414e-b668-f208986d2fd0)

Reply:
This doesn't appear to be resolved with the latest changes (the thick black line at the bottom of the button is incorrect). Change to `new Vector2(215, 151)`